dynamiq/storages/vector/elasticsearch/elasticsearch.py

```diff
@@ -0,0 +1,590 @@
+        self.mapping_settings = mapping_settings or {}
+
+        if create_if_not_exist:
+            self._create_index_if_not_exists()
+
+        logger.debug(f"ElasticsearchVectorStore initialized with index: {self.index_name}")
+
+    def _create_index_if_not_exists(self) -> None:
+        """Create the index if it doesn't exist."""
+        if not self.client.indices.exists(index=self.index_name):
+            # Base mapping
+            mapping = {
+                "mappings": {
+                    "properties": {
+                        self.content_key: {"type": "text"},
+                        "metadata": {"type": "object"},
+                        self.embedding_key: {
+                            "type": "dense_vector",
+                            "dims": self.dimension,
+                            "index": True,
+                            "similarity": self.similarity,
+                        },
+                    }
+                }
+            }
+
+            # Add custom mapping settings if provided
+            if self.mapping_settings:
+                mapping["mappings"].update(self.mapping_settings)
+
+            # Add index settings if provided
+            if self.index_settings:
+                mapping["settings"] = self.index_settings
+
+            self.client.indices.create(index=self.index_name, body=mapping)
+
+    def _handle_duplicate_documents(
+        self, documents: list[Document], policy: DuplicatePolicy = DuplicatePolicy.FAIL
+    ) -> list[Document]:
+        """
+        Handle duplicate documents based on policy.
+
+        Args:
+            documents (list[Document]): List of documents to check for duplicates.
+            policy (DuplicatePolicy): Policy for handling duplicates. Defaults to DuplicatePolicy.FAIL.
+
+        Returns:
+            list[Document]: List of documents after applying the specified policy.
+
+        Raises:
+            VectorStoreException: If duplicates are found and the policy is set to FAIL.
+        """
+        if policy == DuplicatePolicy.OVERWRITE:
+            return documents
+
+        # Get unique documents
+        unique_docs = {}
+        for doc in documents:
+            if doc.id in unique_docs:
+                logger.warning(f"Duplicate document ID found: {doc.id}")
+            unique_docs[doc.id] = doc
+
+        if policy == DuplicatePolicy.SKIP:
+            # Check which documents already exist
+            existing_ids = set()
+            for doc_id in unique_docs.keys():
+                try:
+                    self.client.get(index=self.index_name, id=doc_id)
+                    existing_ids.add(doc_id)
+                except NotFoundError:
+                    pass
+
+            # Remove existing documents
+            return [doc for doc in documents if doc.id not in existing_ids]
+
+        elif policy == DuplicatePolicy.FAIL:
+            # Check for existing documents
+            existing_ids = set()
```

Comment:
This logic is the duplication with DuplicatePolicy.SKIP. The diff if how the `existing_ids` handle after.